lib/src/equatable_utils.dart

```diff
@@ -13,21 +13,7 @@ bool equals(List<Object?> a, List<Object?> b) {
   if (identical(a, b)) return true;
   if (a.length != b.length) return false;
   for (var i = 0; i < a.length; i++) {
-    final unitA = a[i];
-    final unitB = b[i];
-    if (_isEquatable(unitA) && _isEquatable(unitB)) {
-      return unitA == unitB;
-    } else if (unitA is Set && unitB is Set) {
-      return _setEquals(unitA, unitB);
```

Comment:
Same goes for the rest of these --  we can't just return true early without actually checking each unit in the list.